app/src/test/java/org/mozilla/fenix/library/history/HistorySearchControllerTest.kt

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.library.history
+
+import io.mockk.MockKAnnotations
+import io.mockk.impl.annotations.MockK
+import io.mockk.verify
+import kotlinx.coroutines.test.runBlockingTest
+import mozilla.components.concept.engine.EngineSession
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.HomeActivity
+
+class HistorySearchControllerTest {
+
+    @MockK(relaxed = true) private lateinit var activity: HomeActivity
+    @MockK(relaxed = true) private lateinit var store: HistorySearchFragmentStore
+
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this)
+    }
+
+    @Test
+    fun handleEditingCancelled() = runBlockingTest {
+        var clearToolbarFocusInvoked = false
+        createController(
+            clearToolbarFocus = {
+                clearToolbarFocusInvoked = true
+            }
+        ).handleEditingCancelled()
+
+        assertTrue(clearToolbarFocusInvoked)
+    }
+
+    @Test
+    fun handleTextChangedNonEmpty() {
+        val text = "fenix"
+
+        createController().handleTextChanged(text)
+
+        verify { store.dispatch(HistorySearchFragmentAction.UpdateQuery(text)) }
+    }
+
+    @Test
+    fun handleTextChangedEmpty() {
+        val text = ""
+
+        createController().handleTextChanged(text)
+
+        verify { store.dispatch(HistorySearchFragmentAction.UpdateQuery(text)) }
+    }
+
+    @Test
+    fun handleUrlTapped() {
```

Comment:
I've noticed that some tests' names are derivations of the functions they're validating while others are the `WHEN this thing happens THEN this other thing should happen as a result`. What should be the standard practice?

Reply:
I think in general we want to write all new tests using `GIVEN WHEN THEN`, but old habits are tough to break from since it's less verbose.

Reply:
Sounds good.  Bad habit.  I'll fix the test function names.  Thanks